Build/TestingSetupForWebTest.targets

```diff
@@ -29,10 +29,9 @@
     <Error Text="The property 'Platform' is not set." Condition="'$(Platform)' == ''" />
     <Error Text="The property 'AppConfigFile' is not set." Condition="'$(AppConfigFile)' == ''" />
     <Error Text="The property 'Browser' is not set." Condition="'$(Browser)' == ''" />
+    <Error Text="The property 'HostTestSitesInDocker' is not set." Condition="'$(HostTestSitesInDocker)' == ''" />
 
     <PropertyGroup>
-      <_dockerEnabled>False</_dockerEnabled>
-      <_dockerEnabled Condition="'$(DisableDocker)' == 'False'">True</_dockerEnabled>
       <_dockerImageName>mcr.microsoft.com/dotnet/framework/aspnet:4.6.2</_dockerImageName>

```

Comment:
okay, why do we use the 4.6.2 docker image for the docker container that hosts the iis?

Reply:
We'd have to ask @iam-mholle

Reply:
@Stefan-Ilic The build script has the option to select the docker image to use, correct? this information would then be used here. And of course, configured in the webtest-project configration so we test some browsers with .net 4.7.2 iis and some with .net 4.8 iis

Reply:
The TestConfigurations have virtually nothing to do with the IIS hosting and they're basically only concerned with "Am I running nunit.exe inside a docker container or locally, and which TargetFrameworkVersion am I using for running said nunit.exe". In an instance of running a web test, no information regarding a docker image is available since web tests run locally.

Selecting the which docker image should be used for hosting IIS could certainly be added to TestConfigurations, but that would be a separate issue and is not possible ATM.

Off the top of my head, I can also think of solutions that could be done inside the trunk and wouldn't require adapting the BuildScript itself, but that should probably be its own issue as well.

Reply:
Ah, I thought we already have the information as a property?
https://github.com/re-motion/SharedSource-Build/blob/develop/BuildScript/BuildTargets/Testing.targets#L90
` <_dockerImageName>%(_testOutputFiles.ExecutionRuntimeValue)</_dockerImageName>`

Can't we pipe this information to the `_dockerImageName` property in TestingSetupForWebTest.targets?

Reply:
The problem is that if we're currently in a web test run, `_testOutputFiles.ExecutionRuntimeValue` is not a docker image but `LocalMachine` or `EnforcedLocalMachine`

Reply:
The fact that it's still called `_dockerImageName` is probably just an oversight on my part

Reply:
ah, i.c. Fell into the "web test configs are not ordered the same in terms of the target environment" descision. 

Okay, for the immediate future, we can use an ugly workaround by tying the docker image to the browser. that way, we still cover all .net versions in the fullbuild. 

am looking forward to learning to we can just drag the image name along into the test configuration. I know why and how we did things, but is it feasible to add the image name as an extra parameter? this would mean that "EnforcedLocalMachine" becomes a flag instead of an option next to "LocalMachine" and "Win_NET48". Then we could always expose the docker image if it's available (null/blank otherwise). This sounds like the cleanest approach to me but I may be biased and not considering all the options.

Reply:
Tying the docker image and the browser together like `Chrome:Win_NET48` etc. was my first idea for a workaround as well, we'd probably be able to do that without touching the BuildScript and without causing more problems than we solve, but I agree that it wouldnt' be ideal.

We should probably get in a call tomorrow and hash this out, but I would still strongy suggest committing this as is right now so we can get to NUnit3 without falling into a black hole again and create a separate ticket for selecting the IIS hosting environment

Reply:
I was actually thinking even simpler (more hackier) for line 35 in testingSetupForWebtests.targets
` <_dockerImageName Conditon=$(Browser) == "Edge">$(Docker_Win_Net48)</_dockerImageName>`
` <_dockerImageName Conditon=$(Browser) == "Chrome">$(Docker_Win_Net472)</_dockerImageName>`

Yes, it's an awful duplication but it's still only within the local build script and keeps things extremly simple and quick to merge. Design-review for the script extension task has time till next week.